core/src/main/java/hudson/model/Cause.java

```diff
@@ -409,12 +416,15 @@ public String getUserId() {
 
         @Exported(visibility = 3)
         public String getUserName() {
-            return userId == null ? "anonymous" : User.get(userId).getDisplayName();
+            final User user = userId == null ? null : User.getById(userId, false);
+            return user == null ? "anonymous" : user.getDisplayName();
```

Comment:
Perhaps add a `@Nonnull` annotation as we either have "anonymous" or the return a of a nonnull method (User.getDisplayName())

Reply:
Subject for polishing later. Ideally I want to define nonnull requirement on the package level